dynawaltz/src/main/java/com/powsybl/dynawaltz/builders/AbstractEquipmentModelBuilder.java

```diff
@@ -72,7 +74,11 @@ protected void checkData() {
     protected abstract T findEquipment(String staticId);
 
     protected boolean checkEquipment(T equipment) {
-        return Objects.equals(network, equipment.getNetwork());
+        return equipmentPredicates.test(equipment);
```

Comment:
Shouldn't we keep the previous assertion as well?
```suggestion
        return network == equipment.getNetwork() && equipmentPredicates.test(equipment);
```

Reply:
equipmentPredicates is already initialized with the network assertion :
 `private Predicate<T> equipmentPredicates = eq -> Objects.equals(network, eq.getNetwork());`

Reply:
indeed!! missed it